product_sequence/models/product_product.py

```diff
@@ -59,5 +59,10 @@ def copy(self, default=None):
         if default is None:
             default = {}
         if self.default_code and "default_code" not in default:
-            default.update({"default_code": self.default_code + _("-copy")})
+            code = self.default_code
+            while True:  # check for previous duplicates
+                code += _("-copy")
+                if not self.search([("default_code", "=", code)]):
+                    break
+            default.update({"default_code": code})
```

Comment:
We can also prevent that the default_code is also inside the db.


default_code = default.get("default_code", self.default_code)
while self.search([("default_code", "=", default_code)]):
            default_code += _("-copy")
default.update({"default_code": default_code})